dashboard/optional_utils.py

```diff
@@ -35,117 +35,128 @@
 logger = logging.getLogger(__name__)
 
 
-class ClassMethodRouteTable:
-    """A helper class to bind http route to class method."""
-
-    _bind_map = collections.defaultdict(dict)
-    _routes = aiohttp.web.RouteTableDef()
-
-    class _BindInfo:
-        def __init__(self, filename, lineno, instance):
-            self.filename = filename
-            self.lineno = lineno
-            self.instance = instance
-
-    @classmethod
-    def routes(cls):
-        return cls._routes
-
-    @classmethod
-    def bound_routes(cls):
-        bound_items = []
-        for r in cls._routes._items:
-            if isinstance(r, RouteDef):
-                route_method = getattr(r.handler, "__route_method__")
-                route_path = getattr(r.handler, "__route_path__")
-                instance = cls._bind_map[route_method][route_path].instance
-                if instance is not None:
+def method_route_table_factory():
+    class MethodRouteTable:
```

Comment:
This `MethodRouteTable` class is identical to the old `ClassMethodRouteTable`. No code changes were made. It was only renamed and moved into this factory function.